src/resolution_functions/models/model_base.py

```diff
@@ -187,3 +161,58 @@ def citation(self) -> list[str]:
             The citation.
         """
         return self._citation
+
+    def _validate_settings(self,
+                           model_data: ModelData,
+                           settings: dict[str, int | float | None]
+                           ) -> dict[str, int | float]:
+        """
+        Validates the user-provided `settings` against the models restrictions and fills in defaults
+
+        Parameters
+        ----------
+        model_data
+            The data associated with the model for a given version of a given instrument.
+        settings
+            The user-provided :term:`settings<setting>`
+
+        Returns
+        -------
+        validated_settings
+            The user-provided settings that comply with the model's restrictions and with any `None`
+            values replaced with defaults.
+        """
+        out = {}
+        for name, setting in settings.items():
+            if setting is None:
+                try:
+                    out[name] = model_data.defaults[name]
+                except KeyError:
+                    raise InvalidInputError(f'Model "{type(self).__name__}" does not have a default'
+                                            f' value for the "{name}" setting, so one must be '
+                                            f'provided by the user.')
```

Comment:
F-string with no sub
```suggestion
                                            'provided by the user.')
```